modules/heat_conduction/src/postprocessors/ExposedSideAverageValue.C

```diff
@@ -0,0 +1,63 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ExposedSideAverageValue.h"
+#include "SelfShadowSideUserObject.h"
+
+registerMooseObject("MooseApp", ExposedSideAverageValue);
+
+InputParameters
+ExposedSideAverageValue::validParams()
+{
+  InputParameters params = SideAverageValue::validParams();
+  params.addClassDescription("Computes the average value of a variable on the "
+                             "exposed portion of a sideset. Note that this cannot be used on the "
+                             "centerline of an axisymmetric model.");
+  params.addRequiredParam<UserObjectName>(
+      "self_shadow_uo",
+      "SelfShadowSideUserObject that calculates the illumination state of a side set");
+  return params;
+}
+
+ExposedSideAverageValue::ExposedSideAverageValue(const InputParameters & parameters)
+  : SideAverageValue(parameters),
+    _self_shadow(getUserObject<SelfShadowSideUserObject>("self_shadow_uo"))
+{
+  if (_self_shadow.useDisplacedMesh() != getParam<bool>("use_displaced_mesh"))
+    paramWarning("use_displaced_mesh",
+                 "The SelfShadowSideUserObject should operate on the same mesh (displaced or "
+                 "undisplaced) as this PostProcessor.");
+}
+
+Real
+ExposedSideAverageValue::computeQpIntegral()
+{
+  const SelfShadowSideUserObject::SideIDType id(_current_elem->id(), _current_side);
+  const int illumination = _self_shadow.illumination(id);
+  // tests if the bit at position _qp is set
+  if (illumination & (1 << _qp))
+    return SideAverageValue::computeQpIntegral();
+  else
+    return 0.0;
+}
+
+Real
+ExposedSideAverageValue::volume()
+{
+  Real curr_exposed_side_volume = 0.0;
+  const SelfShadowSideUserObject::SideIDType id(_current_elem->id(), _current_side);
+  const int illumination = _self_shadow.illumination(id);
+  for (unsigned int qp = 0; qp < _qrule->n_points(); qp++)
+  {
+    // tests if the bit at position _qp is set
+    if (illumination & (1 << qp))
```

Comment:
https://wiki.sei.cmu.edu/confluence/display/c/INT13-C.+Use+bitwise+operators+only+on+unsigned+operands suggests bitwise operations should only ever be performed on unsigned integers

Reply:
Agreed. I changed `SelfShadowSideUserObject::illumination()` to return an unsigned int, and changed the variable types accordingly.